src/parsing/include/antioch/parser_base.h

```diff
@@ -173,100 +173,103 @@ namespace Antioch
                                                                          >,
-                                                           NumericType > & /*transport_mixture*/)  {not_implemented();}
+                                                           NumericType > & /*transport_mixture*/)  {not_implemented(9);}
 
 
 /// thermo
 
         //! reads the thermo, NASA generalist, no templates for virtual
-        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, NASA7CurveFit<NumericType> >& /*thermo*/)  {not_implemented();}
+        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, NASA7CurveFit<NumericType> >& /*thermo*/)  {not_implemented(10);}
 
         //! reads the thermo, NASA generalist, no templates for virtual
-        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, NASA9CurveFit<NumericType> >& /*thermo*/)  {not_implemented();}
+        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, NASA9CurveFit<NumericType> >& /*thermo*/)  {not_implemented(11);}
 
         //! reads the thermo, NASA generalist, no templates for virtual
-        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, CEACurveFit<NumericType> >& /*thermo*/)  {not_implemented();}
+        virtual void read_thermodynamic_data(NASAThermoMixture<NumericType, CEACurveFit<NumericType> >& /*thermo*/)  {not_implemented(12);}
 
         //! reads the thermo, CEA deprecated
-        virtual void read_thermodynamic_data(CEAThermodynamics<NumericType >& /*thermo*/)  {not_implemented();}
-
+        virtual void read_thermodynamic_data(CEAThermodynamics<NumericType >& /*thermo*/)  {not_implemented(13);}
 
 /// reaction
 
-         /*! go to next reaction*/
-         virtual bool reaction() {not_implemented(); return false;}
+// non const
+
+         /*! read & store current reaction and go to next reaction*/
+         virtual bool reaction() {not_implemented(14); return false;}
 
          /*! go to next rate constant*/
-         virtual bool rate_constant(const std::string & /*kinetics_model*/) {not_implemented(); return false;}
+         virtual bool rate_constant(const std::string & /*kinetics_model*/) {not_implemented(15); return false;}
+
+// const
 
          /*! \return true if there's a Troe block*/
-         virtual bool Troe()  {not_implemented(); return false;}
+         virtual bool Troe() const {not_implemented(16); return false;}
 
          /*! \return reaction id, 0 if not provided*/
-         virtual const std::string reaction_id() const  {not_implemented(); return std::string();}
+         virtual const std::string reaction_id() const  {not_implemented(17); return std::string();}
 
          /*! \return reaction equation */
-         virtual const std::string reaction_equation() const {not_implemented(); return std::string();}
+         virtual const std::string reaction_equation() const {not_implemented(18); return std::string();}
```

Comment:
Shouldn't a lot of these simply be pure virtual functions?  It's much nicer to find out something is not implemented at compile time rather than run time.